src/plugins/intel_gpu/src/graph/space_to_depth.cpp

```diff
@@ -2,16 +2,67 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
+#include "openvino/op/space_to_depth.hpp"
 #include "space_to_depth_inst.h"
 
 #include "primitive_type_base.h"
 #include "intel_gpu/runtime/error_handler.hpp"
 #include "json_object.h"
+#include "data_inst.h"
 #include <string>
 
+#include "space_to_depth_shape_inference.hpp"
+
 namespace cldnn {
 GPU_DEFINE_PRIMITIVE_TYPE_ID(space_to_depth)
 
+template<typename ShapeType>
+std::vector<layout> space_to_depth_inst::calc_output_layouts(space_to_depth_node const& node, kernel_impl_params const& impl_param) {
+    auto desc = impl_param.typed_desc<space_to_depth>();
+    auto input_layout = impl_param.get_input_layout(0);
+
+    auto output_type = desc->output_data_types[0].value_or(input_layout.data_type);
+    if (impl_param.has_fused_primitives())
+        output_type = impl_param.get_fused_output_layout().data_type;
+
+    auto output_format = input_layout.format;
+
+    ov::op::v0::SpaceToDepth op;
+    op.set_block_size(desc->block_size);
+    switch (desc->mode) {
+    case space_to_depth::depth_mode::depth_first:
+        op.set_mode(ov::op::v0::SpaceToDepth::SpaceToDepthMode::DEPTH_FIRST);
+        break;
+    case space_to_depth::depth_mode::blocks_first:
+        op.set_mode(ov::op::v0::SpaceToDepth::SpaceToDepthMode::BLOCKS_FIRST);
+        break;
+    default:
+        CLDNN_ERROR_MESSAGE(desc->id, "Unexpected depth mode " + std::to_string(desc->mode));
+    }
+
+    if (desc->block_size == 0)
+        CLDNN_ERROR_MESSAGE(desc->id, "Invalid block size " + std::to_string(desc->block_size));
+
+    if (input_layout.is_static()) {
+        const size_t spatial_num = format::spatial_num(input_layout.format);
```

Comment:
I think we can remove these checks from our side, because they are implemented in shape_infer() function itself

Reply:
Removed.